jcstress-samples/src/main/java/org/openjdk/jcstress/samples/concurrency/classic/ClassicProblem_02_ProducerConsumerProblem.java

```diff
@@ -0,0 +1,316 @@
+
+        @Actor
+        void producer1(III_Result r) {
+            r.r1 = produce();
+        }
+
+        @Actor
+        void producer2(III_Result r) {
+            r.r2 = produce();
+        }
+
+        @Actor
+        void consume(III_Result r) {
+            r.r3 = consume();
+        }
+
+        @Override
+        protected int putItemIntoBuffer() {
+            synchronized (indexLock) {
+                return index++;
+            }
+        }
+
+        @Override
+        protected void takeItemFromBuffer() {
+            synchronized (indexLock) {
+                index--;
+            }
+        }
+    }
+
+    /*
+        This solution with a ReentrantLock and two conditions works with many producers and many consumers.
+     */
+    @JCStressTest
+    @Outcome(expect = ACCEPTABLE, desc = "Producers didn't overwrite each other's item.")
+    @Outcome(id = {"0, 0, 2"}, expect = ACCEPTABLE_INTERESTING, desc = "Producers overwrote each other's item.")
+    @State
+    public static class Lock {
+        private final ReentrantLock lock = new ReentrantLock();
+        private final Condition full = lock.newCondition();
+        private final Condition empty = lock.newCondition();
+        private int count = 0;
+
+        @Actor
+        void producer1(III_Result r) {
+            r.r1 = produce();
+        }
+
+        @Actor
+        void producer2(III_Result r) {
+            r.r2 = produce();
+        }
+
+        @Actor
+        void consumer(III_Result r) {
+            r.r3 = consume();
+        }
+
+        public int produce() {
+            // produce item
+            lock.lock();
+            try {
+                while(count == BUFFER_SIZE) {
+                    full.await();
+                }
+                // put item to buffer
+                int index = count++;
+                if(count == 1) {
+                    empty.signalAll();
+                }
+                return index;
+            } catch (InterruptedException e) {
+                throw new IllegalStateException(e);
+            } finally {
+                lock.unlock();
+            }
+        }
+
+        public int consume() {
+            int result;
+            lock.lock();
+            try {
+                while(count == 0) {
+                    empty.await();
+                }
+                result = count;
+                // put item to buffer
+                count--;
+                if(count == BUFFER_SIZE - 1) {
+                    full.signalAll();
+                }
+            } catch (InterruptedException e) {
+                throw new IllegalStateException(e);
+            } finally {
+                lock.unlock();
+            }
+            // consume item
+            return result;
+        }
+    }
+
+    /*
+        This solution with AtomicIntegers only works with one producer and one consumer.
+        It fails if an int overflow happens.
+     */
+    @SuppressWarnings("StatementWithEmptyBody")
```

Comment:
We don't do `@SuppressWarnings` in jcstress code, I think. At least not in the samples code.